### PR TITLE
Add LGTM.com C/C++ analysis configuration and badges (#138)

### PR DESCRIPTION
**Pull request by sjvs**
Hi there!

I noticed that you're interested in LGTM.com (https://github.com/utilForever/Hearthstonepp/issues/138) and thought I'd give you a hand with writing an `lgtm.yml` to get analysis up and running for your C/C++ code.

The first results are already available on LGTM.com: https://lgtm.com/projects/g/utilForever/Hearthstonepp/alerts

Note that the vast majority of alerts is actually in library code; the `lgtm.yml` file classifies that code accordingly so you won't be bothered by those results (and so they don't affect your code quality score!).

I've also included some badges. As soon as this PR is merged and the library code is classified correctly, your code quality grade will significantly improve.

Any questions: give me a shout. Full disclosure: I'm on the LGTM.com team.

### CODE REVIEW

**Comment by sjvs** [comment]
Oh, and you may want to consider enabling LGTM's automated code review for pull requests: https://lgtm.com/projects/g/utilForever/Hearthstonepp/ci/

It'll automatically post a comment on a PR if it introduces (or fixes :slightly_smiling_face:) an alert. Here's an example of how that works for Google's AMPHTML project: https://github.com/ampproject/amphtml/pull/13060

**Comment by sjvs** [comment]
cc @utilForever — in case you didn't get notified about this :slightly_smiling_face:

**Comment by sjvs** [comment]
I've just added a commit that should make sure that LGTM analysis your code as Python 3, rather than 2.

**Comment by utilForever** [comment]
@sjvs Hi! First of all, thank you for your work. I saw your commit. I noticed that there are a lot of warnings in third-party libraries. Could you ignore it? Maybe ```Libraries/``` folder. Also, ```Tests/``` contains unit test code.

**Comment by sjvs** [comment]
I've disabled scanning of test code altogether with a `cmake` flag (see `lgtm.yml`). The `Libraries` directory will be classified properly as soon as this PR is merged: `lgtm.yml` contains the right classification hints for LGTM.

So I think everything should work with this :slightly_smiling_face:

File: README.md [hunk 1]
```diff
@@ -173,4 +175,4 @@ Permission is hereby granted, free of charge, to any person obtaining a copy of
 
 The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
\ No newline at end of file
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
[the remark this answers is not in the document]
Comment (utilForever) [reply README.md]: Oh... Don't mind. :smile:

**Comment by utilForever** [comment]
@sjvs Again, thanks for your work! I'll continue to work on it, and please promote it. 🙏

**Comment by sjvs** [comment]
Thanks for merging @utilForever! And let me know if you have any suggestions/comments/questions about LGTM. Happy to help!

Don't forget to have a look at the automatic code review integration I mentioned above. If you like, you can enable it here: https://lgtm.com/projects/g/utilForever/Hearthstonepp/ci/

**Pull request merged**